src/main/java/com/github/games647/flexiblelogin/commands/admin/AccountsCommand.java

```diff
@@ -0,0 +1,149 @@
+
+    @Inject
+    private NamePredicate namePredicate;
+
+    @Inject
+    private IPv4Predicate ipv4Predicate;
+
+    @Inject
+    AccountsCommand(FlexibleLogin plugin, Logger logger, Settings settings) {
+        super(plugin, logger, settings);
+    }
+
+    @Override
+    public CommandResult execute(CommandSource src, CommandContext args) throws CommandException {
+        String accountId = args.<String>getOne("account").get();
+
+        if (ipv4Predicate.test(accountId)) {
+            Task.builder()
+                    //we are executing a SQL Query which is blocking
+                    .async()
+                    .execute(() -> {
+                        InetAddress ip = null;
+                        try {
+                            ip = InetAddress.getByName(accountId);
+                        } catch (UnknownHostException ex) {
+                            src.sendMessage(plugin.getConfigManager().getText().getInvalidIPv4());
+                            return;
+                        }
+                        
+                        Set<Account> accounts = plugin.getDatabase().getAccountsByIp(ip);
+                        sendAccountsList(src, accountId, accounts);
+                    })
+                    .submit(plugin);
+            return CommandResult.success();
+        } else if (namePredicate.test(accountId)) {
+            Task.builder()
+                    //we are executing a SQL Query which is blocking
+                    .async()
+                    .execute(() -> {
+                        Optional<Account> optAccount = plugin.getDatabase().loadAccount(accountId);
+                        if(!optAccount.isPresent()) {
+                            src.sendMessage(plugin.getConfigManager().getText().getAccountNotFound());
+                            return;
+                        }
+                        
+                        InetAddress ip = optAccount.get().getIP();
+                        
+                        if(ip == null) {
+                            src.sendMessage(plugin.getConfigManager().getText().getAccountsListNoIP());
+                            return;
+                        }
+                        
+                        Set<Account> accounts = plugin.getDatabase().getAccountsByIp(ip);
+                        
+                        sendAccountsList(src, accountId, accounts);
+                    })
+                    .submit(plugin);
+            return CommandResult.success();
+        }
+
+        src.sendMessage(settings.getText().getInvalidUsername());
+        return CommandResult.success();
+    }
+
+    private void sendAccountsList(CommandSource src, String username, Set<Account> accounts) {
+        if(accounts.isEmpty()) {
+            src.sendMessage(plugin.getConfigManager().getText().getAccountsListEmpty());
+            return;
+        }
+        
+        List<String> accountsList = new ArrayList<>();
```

Comment:
(Not required) Note this could be simplified with streams. Something similar like

```java
List<String> names = accounts.stream.map(Account::getUsername)
.filter(Optional::isPresent)
.map(Optional::get)
.collect(Collectors.toList();
```